### PR TITLE
[ORCA-279] Modular Validation and Scoring

### PR DESCRIPTION
**Pull request by BWMac**
This PR follows on from #1

In order to support multiple individual challenges within our challenge types, we need to allow for different validation and scoring methods to be used. This PR adds support for this by adding `scoring_script` and `validation_script` parameters to each subworkflow in the `SCORE` and `VALIDATE` steps. Each parameter is a string with the name of the scoring or validation script to be used, which is then passed on to the command executed in each process. The parameters default to our placeholder scripts that are already within the `bin` directory. I have also updated `README.md` to reflect how this changes the process of contributing new challenges and/or scoring/validation methods.

These changes have been tested both locally and on Tower:

- [Model-to-Data](https://tower-dev.sagebionetworks.org/orgs/Sage-Bionetworks/workspaces/example-dev-project/watch/CV130MIYPOXjc)
- [Data-to-Model](https://tower-dev.sagebionetworks.org/orgs/Sage-Bionetworks/workspaces/example-dev-project/watch/wThfgFHjbHR6A)

### CODE REVIEW

File: modules/validate.nf [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
     val ready
+    val validation_script
 
     output:
     tuple val(submission_id), path(predictions), stdout, path("results.json")
 
     script:
     """
-    validate.py '${predictions}'
+    ${validation_script} '${predictions}'
```
Comment (thomasyu888) [inline modules/validate.nf:16]: Ah, I see.  I think we should support different containers, but limit the inputs and outputs of that container.

I can imagine challenge organizers wanting to build their own validation and scoring container per challenge

**Review by thomasyu888 — APPROVED** [review]
🔥 I'm going to pre-approve, but will you include the scoring and validation python scripts in this PR?

Let's just actively thinking about the fact that the folder structure can change a little bit as we add challenges

**Comment by BWMac** [comment]
> but will you include the scoring and validation python scripts in this PR?

@thomasyu888 Which scripts do I need to include? I already have the placeholder ones we have been using here, but I am not aware of any further scoring or validation scripts.

**Comment by thomasyu888** [comment]
@BWMac ah, I just realized you forked this repo from an existing repo so it contained the scripts already.  You can ignore my comment

**Pull request merged**